### PR TITLE
Revise product version

### LINKED ISSUE #4316: Fix the timestamp in the version string
**Describe the bug**
The version string shows the current date instead of the commit date. Thus, it shows a different version string each day (if restarted).

**To Reproduce**
Run Nethermind with the `--version` option.

**Expected behavior**
The version string must be the same for a given version regardless of the date.

**Screenshots**
image

### PR DESCRIPTION
**Pull request by rubo**
Fixes #4316

## Changes:
**Version change doesn't require changing Git tagging. Also, it no longer relies on the `git-hash` file, so the file has gone. All the version-related info is now embedded into the assemblies.**

The product version is now [semantic](https://semver.org/). For the `build` label, an 8-digit commit hash is used. For instance, `1.13.5+c8a4bc95`. The redundant git-related info has been removed from the version.

~The client id is formatted similar to web browsers' user agent format which is well known to most users:
`Nethermind/1.13.5+c8a4bc95 (Linux; x64; .NET 6.0.7)`~

The client id is revised: `Nethermind/v1.13.5+5a619799/linux-x64/dotnet6.0.7`

Some other Ethereum clients to compare to:
- `Geth/v1.10.20-stable-8f2416a8/windows-amd64/go1.18.1`
- `besu/v22.4.2-dev-7ea97fd5/linux-x86_64/openjdk-java-11`
- `erigon/v2022.07.2-alpha-c7a94eee/linux-amd64/go1.18.1`
- `OpenEthereum/v3.3.0-rc.4-unstable-745c4bd-20210803/x86_64-linux-gnu/rustc1.51.0`
- `nimbus-eth1 v0.1.0 [linux: amd64, rocksdb, nimvm, 6cfaaf]`

With the `--version` option:
```
Version: 1.13.5+d2760bed
Commit: d2760bed1012e4a39aa40784d79cf35063325d58
Build Date: 2022-07-27 15:42:16Z
OS: Linux x64
Runtime: .NET 6.0.7
```
This is more similar to what Geth does with the same option.

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [x] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] Other (please describe): 

## Testing
**Requires testing**

- [x] Yes
- [ ] No

**In case you checked yes, did you write tests??**

- [x] Yes
- [ ] No

## Further comments (optional)

A lot of files have been modified because of the `ClientVersion` class rename and file-scoped namespaces. No logic changes.

### CODE REVIEW

**Review by AntiD2ta — APPROVED** [review]
TL;DR 
LGTM

I checked only the workflow files and they seems good to go. @rubo changes require the latest tag as a release version, so he need the entire history (`fetch-depth: 0`) to be able to get that commit. We may came up with a better solution at some point but for now I don't see any major issue with it.

**Review by LukaszRozmej — COMMENTED** [review]
Please fix the conflict.

Will it work just build from solution without the pipeline?

**Comment by rubo** [comment]
@LukaszRozmej Yes, of course. It will behave similarly to what it does now showing the _latest_ commit and the _latest_ tag suffixed with `-dirty` if the repo is dirty. For example, `1.13.5-dirty+d2760bed`.

**Review by FalcoXYZ — APPROVED** [review]
Only checked workflow files/scripts, LGTM.
